File: BankingSystem.py
import pickle
import os
import getpass
import time

class BankAccount:
    def __init__(self, accountNumber, accountHolderName, pin):
        self.accountNumber = accountNumber
        self.accountHolderName = accountHolderName
        self.balance = 0.0
        self.loan = 0.0
        self.pin = pin

    def getAccountNumber(self):
        return self.accountNumber

    def deposit(self, amount):
        self.balance += amount
        print("Deposit successful.")
        print("Current balance: $" + str(self.balance))

    def withdraw(self, amount):
        if self.balance + self.loan >= amount:
            if self.balance >= amount:
                self.balance -= amount
                print("Withdrawal successful.")
                print("Current balance: $" + str(self.balance))
            else:
                remainingAmount = amount - self.balance
                self.balance = 0
                self.loan -= remainingAmount
                print("Withdrawal successful.")
                print("Current balance: $" + str(self.balance))
                print("Loan amount remaining: $" + str(self.loan))
        else:
            print("Insufficient funds.")
            print("Current balance: $" + str(self.balance))

    def displayAccountDetails(self):
        print("Account Number: " + self.accountNumber)
        print("Account Holder: " + self.accountHolderName)
        print("Current Balance: $" + str(self.balance))
        print("Loan Amount: $" + str(self.loan))

    def requestLoan(self, loanAmount):
        if loanAmount > 0:
            self.loan += loanAmount
            interest = loanAmount * (5.0 / 100)  # 5% interest rate
            self.balance += loanAmount - interest
            print("Loan approved.")
            print("Loan amount: $" + str(loanAmount))
            print("Loan interest: $" + str(interest))
            print("Current balance: $" + str(self.balance))
        else:
            print("Invalid loan request.")

    def repayLoan(self, amount):
        if self.loan > 0:
            if amount >= self.loan:
                self.balance -= amount
                print("Loan fully repaid.")
                print("Current balance: $" + str(self.balance))
                print("Loan amount remaining: $0.00")
                self.loan = 0
            else:
                self.balance -= amount
                self.loan -= amount
                print("Loan partially repaid.")
                print("Current balance: $" + str(self.balance))
                print("Loan amount remaining: $" + str(self.loan))
        else:
            print("No active loan.")

    def verifyPin(self, enteredPin):
        if len(enteredPin) != len(self.pin):
            return False
        for i in range(len(self.pin)):
            if enteredPin[i] != self.pin[i]:
                return False
        return True

    def payBill(self, billType, amount):
        if billType == "Meralco":
            # Process Meralco bill payment
            if self.balance >= amount:
                self.balance -= amount
                print(f"{billType} bill payment of ${amount} successful.")
                print("Current balance: $" + str(self.balance))
            else:
                print("Insufficient funds.")
                print("Current balance: $" + str(self.balance))
        elif billType == "Water District":
            # Process Water District bill payment
            if self.balance >= amount:
                self.balance -= amount
                print(f"{billType} bill payment of ${amount} successful.")
                print("Current balance: $" + str(self.balance))
            else:
                print("Insufficient funds.")
                print("Current balance: $" + str(self.balance))
        elif billType == "Converge":
            # Process Converge bill payment
            if self.balance >= amount:
                self.balance -= amount
                print(f"{billType} bill payment of ${amount} successful.")
                print("Current balance: $" + str(self.balance))
            else:
                print("Insufficient funds.")
                print("Current balance: $" + str(self.balance))
        elif billType == "PLDT":
            # Process PLDT bill payment
            if self.balance >= amount:
                self.balance -= amount
                print(f"{billType} bill payment of ${amount} successful.")
                print("Current balance: $" + str(self.balance))
            else:
                print("Insufficient funds.")
                print("Current balance: $" + str(self.balance))
        else:
            print("Invalid bill type.")

class Bank:
    def __init__(self):
        self.accounts = []
        self.DATA_FILE = "accounts.pkl"
        self.loadAccounts()

    def createAccount(self, accountNumber, accountHolderName, pin):
        print("Creating account...")
        self.loadingScreen()

        account = BankAccount(accountNumber, accountHolderName, pin)
        self.accounts.append(account)

        print("Account created successfully.")
        self.loadingScreen()

        self.saveAccounts()

    def findAccount(self, accountNumber):
        for account in self.accounts:
            if account.getAccountNumber() == accountNumber:
                return account
        return None

    def requestLoan(self, accountNumber, enteredPin, loanAmount):
        loanAccount = self.findAccount(accountNumber)
        if loanAccount:
            if loanAccount.verifyPin(enteredPin):
                loanAccount.requestLoan(loanAmount)
                self.saveAccounts()
                self.loadingScreen()
            else:
                print("Invalid PIN.")
                self.loadingScreen()
        else:
            print("Account not found.")
            self.loadingScreen()

    def loadAccounts(self):
        try:
            with open(self.DATA_FILE, 'rb') as file:
                self.accounts = pickle.load(file)
                print("Accounts loaded successfully.")
        except (IOError, EOFError):
            print("No existing accounts found.")

    def saveAccounts(self):
        try:
            with open(self.DATA_FILE, 'wb') as file:
                pickle.dump(self.accounts, file)
                print("Accounts saved successfully.")
        except IOError:
            print("Failed to save accounts.")

    def payBill(self, accountNumber, enteredPin, billType, amount):
        billAccount = self.findAccount(accountNumber)
        if billAccount:
            if billAccount.verifyPin(enteredPin):
                billAccount.payBill(billType, amount)
                self.saveAccounts()
                self.loadingScreen()
            else:
                print("Invalid PIN.")
                self.loadingScreen()
        else:
            print("Account not found.")
            self.loadingScreen()

    @staticmethod
    def loadingScreen():
        time.sleep(2)  # Adjust the duration of the loading screen as needed

def clearScreen():
    os.system('cls' if os.name == 'nt' else 'clear')

def main():
    bank = Bank()

    while True:
        clearScreen()
        print("   _____ __              __             _    __            ____")
        print("  / ___// /_  ____ _____/ /___ _      _| |  / /___ ___  __/ / /_")
        print("  \\__ \\/ __ \\/ __ `/ __  / __ \\ | /| / / | / / __ `/ / / / / __/")
        print(" ___/ / / / / /_/ / /_/ / /_/ / |/ |/ /| |/ / /_/ / /_/ / / /_")
        print("/____/_/ /_/\\__,_/\\__,_/\\____/|__/|__/ |___/\\__,_/\\__,_/_/\\__/")
        print("                                                By veilwr4ith            ")
        print("\nWelcome to the Banking System!")
        print("1. Create Account")
        print("2. Deposit")
        print("3. Withdraw")
        print("4. Account Details")
        print("5. Request Loan")
        print("6. Repay Loan")
        print("7. Pay Bills")
        print("8. Exit")

        choice = input("\nEnter your choice: ")

        if choice == '1':
            print("\nCreating Account")
            print("-----------------")
            accountNumber = input("Enter Account Number: ")
            accountHolderName = input("Enter Account Holder Name: ")
            pin = getpass.getpass("Set PIN: ")
            bank.createAccount(accountNumber, accountHolderName, pin)
            input("\nPress Enter to continue...")
            continue
        elif choice == '2':
            print("\nDeposit")
            print("---------------")
            accountNumber = input("Enter Account Number: ")
            depositAccount = bank.findAccount(accountNumber)
            if depositAccount:
                enteredPin = getpass.getpass("Enter PIN: ")
                if depositAccount.verifyPin(enteredPin):
                    depositAmount = float(input("Enter Deposit Amount: "))
                    print("---------------")
                    depositAccount.deposit(depositAmount)
                    bank.saveAccounts()
                    bank.loadingScreen()
                    input("\nPress Enter to continue...")
                    continue
                else:
                    print("Invalid PIN.")
                    bank.loadingScreen()
                    input("\nPress Enter to continue...")
                    continue
            else:
                print("Account not found.")
                bank.loadingScreen()
                input("\nPress Enter to continue...")
                continue
        elif choice == '3':
            print("\nWithdraw")
            print("---------------")
            accountNumber = input("Enter Account Number: ")
            withdrawAccount = bank.findAccount(accountNumber)
            if withdrawAccount:
                enteredPin = getpass.getpass("Enter PIN: ")
                if withdrawAccount.verifyPin(enteredPin):
                    withdrawalAmount = float(input("Enter Withdrawal Amount: "))
                    print("---------------")
                    withdrawAccount.withdraw(withdrawalAmount)
                    bank.saveAccounts()
                    bank.loadingScreen()
                    input("\nPress Enter to continue...")
                    continue
                else:
                    print("Invalid PIN.")
                    bank.loadingScreen()
                    input("\nPress Enter to continue...")
                    continue
            else:
                print("Account not found.")
                bank.loadingScreen()
                input("\nPress Enter to continue...")
                continue
        elif choice == '4':
            print("\nAccount Details")
            print("---------------")
            accountNumber = input("Enter Account Number: ")
            accountDetailsAccount = bank.findAccount(accountNumber)
            if accountDetailsAccount:
                enteredPin = getpass.getpass("Enter PIN: ")
                print("---------------")
                if accountDetailsAccount.verifyPin(enteredPin):
                    accountDetailsAccount.displayAccountDetails()
                    bank.loadingScreen()
                    input("\nPress Enter to continue...")
                    continue
                else:
                    print("Invalid PIN.")
                    bank.loadingScreen()
                    input("\nPress Enter to continue...")
                    continue
            else:
                print("Account not found.")
                bank.loadingScreen()
                input("\nPress Enter to continue...")
                continue
        elif choice == '5':
            print("\nRequest Loan")
            print("---------------")
            accountNumber = input("Enter Account Number: ")
            loanAccount = bank.findAccount(accountNumber)
            if loanAccount:
                enteredPin = getpass.getpass("Enter PIN: ")
                if loanAccount.verifyPin(enteredPin):
                    loanAmount = float(input("Enter Loan Amount: "))
                    print("---------------")
                    bank.requestLoan(accountNumber, enteredPin, loanAmount)
                    input("\nPress Enter to continue...")
                    continue
                else:
                    print("Invalid PIN.")
                    bank.loadingScreen()
                    input("\nPress Enter to continue...")
                    continue
            else:
                print("Account not found.")
                bank.loadingScreen()
                input("\nPress Enter to continue...")
                continue
        elif choice == '6':
            print("\nRepay Loan")
            print("---------------")
            accountNumber = input("Enter Account Number: ")
            repayLoanAccount = bank.findAccount(accountNumber)
            if repayLoanAccount:
                enteredPin = getpass.getpass("Enter PIN: ")
                if repayLoanAccount.verifyPin(enteredPin):
                    repaymentAmount = float(input("Enter Repayment Amount: "))
                    print("---------------")
                    repayLoanAccount.repayLoan(repaymentAmount)
                    bank.saveAccounts()
                    bank.loadingScreen()
                    input("\nPress Enter to continue...")
                    continue
                else:
                    print("Invalid PIN.")
                    bank.loadingScreen()
                    input("\nPress Enter to continue...")
                    continue
            else:
                print("Account not found.")
                bank.loadingScreen()
                input("\nPress Enter to continue...")
                continue
        if choice == '7':
            print("\nPay Bill")
            print("---------------")
            accountNumber = input("Enter Account Number: ")
            billAccount = bank.findAccount(accountNumber)
            if billAccount:
                enteredPin = getpass.getpass("Enter PIN: ")
                if billAccount.verifyPin(enteredPin):
                    print("Select bill type:")
                    print("1. Meralco")
                    print("2. Water District")
                    print("3. Converge")
                    print("4. PLDT")
                    billChoice = input("Enter your choice: ")

                    if billChoice == '1':
                        billType = "Meralco"
                        amount = float(input("Enter Meralco bill amount: "))
                    elif billChoice == '2':
                        billType = "Water District"
                        amount = float(input("Enter Water District bill amount: "))
                    elif billChoice == '3':
                        billType = "Converge"
                        amount = float(input("Enter Converge bill amount: "))
                    elif billChoice == '4':
                        billType = "PLDT"
                        amount = float(input("Enter PLDT bill amount: "))
                    else:
                        print("Invalid bill choice.")
                        continue

                    print("---------------")
                    bank.payBill(accountNumber, enteredPin, billType, amount)
                    input("\nPress Enter to continue...")
                    continue
                else:
                    print("Invalid PIN.")
                    bank.loadingScreen()
                    input("\nPress Enter to continue...")
                    continue
            else:
                print("Account not found.")
                bank.loadingScreen()
                input("\nPress Enter to continue...")
                continue

        elif choice == '8':
            print("\nExiting the program. Thank you for using the Banking System!")
            break
        else:
            print("Invalid choice. Please try again.")
            bank.loadingScreen()

if __name__ == "__main__":
    main()

File: BankingSystem.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

class BankAccount implements Serializable {
    private String accountNumber;
    private String accountHolderName;
    private double balance;
    private double loan;
    private char[] pin;

    public BankAccount(String accountNumber, String accountHolderName, char[] pin) {
        this.accountNumber = accountNumber;
        this.accountHolderName = accountHolderName;
        this.balance = 0.0;
        this.loan = 0.0;
        this.pin = pin;
    }

    public String getAccountNumber() {
        return accountNumber;
    }

    // Getter and setter methods

    public void deposit(double amount) {
        balance += amount;
        System.out.println("Deposit successful.");
        System.out.println("Current balance: $" + balance);
    }

    public void withdraw(double amount) {
        if (balance + loan >= amount) {
            if (balance >= amount) {
                balance -= amount;
                System.out.println("Withdrawal successful.");
                System.out.println("Current balance: $" + balance);
            } else {
                double remainingAmount = amount - balance;
                balance = 0;
                loan -= remainingAmount;
                System.out.println("Withdrawal successful.");
                System.out.println("Current balance: $" + balance);
                System.out.println("Loan amount remaining: $" + loan);
            }
        } else {
            System.out.println("Insufficient funds.");
            System.out.println("Current balance: $" + balance);
        }
    }

    public void displayAccountDetails() {
        System.out.println("Account Number: " + accountNumber);
        System.out.println("Account Holder: " + accountHolderName);
        System.out.println("Current Balance: $" + balance);
        System.out.println("Loan Amount: $" + loan);
    }

    public void requestLoan(double loanAmount) {
    if (loanAmount > 0) {
        loan += loanAmount;
        double interest = loanAmount * (5.0 / 100); // 5% interest rate
        balance += loanAmount - interest;
        System.out.println("Loan approved.");
        System.out.println("Loan amount: $" + loanAmount);
        System.out.println("Loan interest: $" + interest);
        System.out.println("Current balance: $" + balance);
    } else {
        System.out.println("Invalid loan request.");
    }
}

    public void repayLoan(double amount) {
        if (loan > 0) {
            if (amount >= loan) {
                balance -= amount;
                System.out.println("Loan fully repaid.");
                System.out.println("Current balance: $" + balance);
                System.out.println("Loan amount remaining: $0.00");
                loan = 0;
            } else {
                balance -= amount;
                loan -= amount;
                System.out.println("Loan partially repaid.");
                System.out.println("Current balance: $" + balance);
                System.out.println("Loan amount remaining: $" + loan);
            }
        } else {
            System.out.println("No active loan.");
        }
    }

    public boolean verifyPin(char[] enteredPin) {
        if (enteredPin.length != pin.length) {
            return false;
        }
        for (int i = 0; i < pin.length; i++) {
            if (enteredPin[i] != pin[i]) {
                return false;
            }
        }
        return true;
    }
}

class Bank implements Serializable {
    private List<BankAccount> accounts;
    private static final String DATA_FILE = "accounts.dat";

    public Bank() {
        accounts = new ArrayList<>();
        loadAccounts();
    }

    public void createAccount(String accountNumber, String accountHolderName, char[] pin) {
        System.out.println("Creating account...");
        loadingScreen();

        BankAccount account = new BankAccount(accountNumber, accountHolderName, pin);
        accounts.add(account);

        System.out.println("Account created successfully.");
        loadingScreen();

        saveAccounts();
    }

    public BankAccount findAccount(String accountNumber) {
        for (BankAccount account : accounts) {
            if (account.getAccountNumber().equals(accountNumber)) {
                return account;
            }
        }
        return null;
    }

    public void requestLoan(String accountNumber, char[] enteredPin, double loanAmount) {
        BankAccount loanAccount = findAccount(accountNumber);
        if (loanAccount != null) {
            if (loanAccount.verifyPin(enteredPin)) {
                loanAccount.requestLoan(loanAmount);
                saveAccounts();
                Bank.loadingScreen();
            } else {
                System.out.println("Invalid PIN.");
                Bank.loadingScreen();
            }
        } else {
            System.out.println("Account not found.");
            Bank.loadingScreen();
        }
    }

    private void loadAccounts() {
        try (ObjectInputStream inputStream = new ObjectInputStream(new FileInputStream(DATA_FILE))) {
            accounts = (List<BankAccount>) inputStream.readObject();
            System.out.println("Accounts loaded successfully.");
        } catch (IOException | ClassNotFoundException e) {
            System.out.println("No existing accounts found.");
        }
    }

    public void saveAccounts() {
        try (ObjectOutputStream outputStream = new ObjectOutputStream(new FileOutputStream(DATA_FILE))) {
            outputStream.writeObject(accounts);
            System.out.println("Accounts saved successfully.");
        } catch (IOException e) {
            System.out.println("Failed to save accounts.");
        }
    }

    public static void loadingScreen() {
        try {
            Thread.sleep(2000); // Adjust the duration of the loading screen as needed
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

public class BankingSystem {
    public static void main(String[] args) {
        Bank bank = new Bank();
        Scanner scanner = new Scanner(System.in);
        int choice = 0;

        while (choice != 7) {
            clearScreen();
            System.out.println("   _____ __              __             _    __            ____");
            System.out.println("  / ___// /_  ____ _____/ /___ _      _| |  / /___ ___  __/ / /_");
            System.out.println("  \\__ \\/ __ \\/ __ `/ __  / __ \\ | /| / / | / / __ `/ / / / / __/");
            System.out.println(" ___/ / / / / /_/ / /_/ / /_/ / |/ |/ /| |/ / /_/ / /_/ / / /_");
            System.out.println("/____/_/ /_/\\__,_/\\__,_/\\____/|__/|__/ |___/\\__,_/\\__,_/_/\\__/");
            System.out.println("                                                By veilwr4ith            ");
            System.out.println("\nWelcome to the Banking System!");
            System.out.println("1. Create Account");
            System.out.println("2. Deposit");
            System.out.println("3. Withdraw");
            System.out.println("4. Account Details");
            System.out.println("5. Request Loan");
            System.out.println("6. Repay Loan");
            System.out.println("7. Exit");
            System.out.print("\nEnter your choice: ");
            choice = scanner.nextInt();
            scanner.nextLine(); // Consume the newline character

            clearScreen();
            switch (choice) {
                case 1:
                    System.out.println("   _____ __              __             _    __            ____");
                    System.out.println("  / ___// /_  ____ _____/ /___ _      _| |  / /___ ___  __/ / /_");
                    System.out.println("  \\__ \\/ __ \\/ __ `/ __  / __ \\ | /| / / | / / __ `/ / / / / __/");
                    System.out.println(" ___/ / / / / /_/ / /_/ / /_/ / |/ |/ /| |/ / /_/ / /_/ / / /_");
                    System.out.println("/____/_/ /_/\\__,_/\\__,_/\\____/|__/|__/ |___/\\__,_/\\__,_/_/\\__/");
                    System.out.println("                                                By veilwr4ith           ");
                    System.out.println("\nCreating Account");
                    System.out.println("-----------------");
                    System.out.print("Enter Account Number: ");
                    String accountNumber = scanner.nextLine();
                    System.out.print("Enter Account Holder Name: ");
                    String accountHolderName = scanner.nextLine();
                    System.out.print("Set PIN: ");
                    char[] pin = System.console().readPassword(); // Read the PIN without echoing
                    bank.createAccount(accountNumber, accountHolderName, pin);
                    break;
                case 2:
                    System.out.println("   _____ __              __             _    __            ____");
                    System.out.println("  / ___// /_  ____ _____/ /___ _      _| |  / /___ ___  __/ / /_");
                    System.out.println("  \\__ \\/ __ \\/ __ `/ __  / __ \\ | /| / / | / / __ `/ / / / / __/");
                    System.out.println(" ___/ / / / / /_/ / /_/ / /_/ / |/ |/ /| |/ / /_/ / /_/ / / /_");
                    System.out.println("/____/_/ /_/\\__,_/\\__,_/\\____/|__/|__/ |___/\\__,_/\\__,_/_/\\__/");
                    System.out.println("                                                By veilwr4ith             ");
                    System.out.println("\nDeposit");
                    System.out.println("---------------");
                    System.out.print("Enter Account Number: ");
                    accountNumber = scanner.nextLine();
                    BankAccount depositAccount = bank.findAccount(accountNumber);
                    if (depositAccount != null) {
                        System.out.print("Enter PIN: ");
                        char[] enteredPin = System.console().readPassword(); // Read the PIN without echoing
                        if (depositAccount.verifyPin(enteredPin)) {
                            System.out.print("Enter Deposit Amount: ");
                            double depositAmount = scanner.nextDouble();
                            scanner.nextLine(); // Consume the newline character
                            System.out.println("---------------");
                            depositAccount.deposit(depositAmount);
                            bank.saveAccounts(); // Save account balances after deposit
                            Bank.loadingScreen();
                        } else {
                            System.out.println("Invalid PIN.");
                            Bank.loadingScreen();
                        }
                    } else {
                        System.out.println("Account not found.");
                        Bank.loadingScreen();
                    }
                    break;
                case 3:
                    System.out.println("   _____ __              __             _    __            ____");
                    System.out.println("  / ___// /_  ____ _____/ /___ _      _| |  / /___ ___  __/ / /_");
                    System.out.println("  \\__ \\/ __ \\/ __ `/ __  / __ \\ | /| / / | / / __ `/ / / / / __/");
                    System.out.println(" ___/ / / / / /_/ / /_/ / /_/ / |/ |/ /| |/ / /_/ / /_/ / / /_");
                    System.out.println("/____/_/ /_/\\__,_/\\__,_/\\____/|__/|__/ |___/\\__,_/\\__,_/_/\\__/");
                    System.out.println("                                                By veilwr4ith            ");
                    System.out.println("\nWithdraw");
                    System.out.println("---------------");
                    System.out.print("Enter Account Number: ");
                    accountNumber = scanner.nextLine();
                    BankAccount withdrawAccount = bank.findAccount(accountNumber);
                    if (withdrawAccount != null) {
                        System.out.print("Enter PIN: ");
                        char[] enteredPin = System.console().readPassword(); // Read the PIN without echoing
                        if (withdrawAccount.verifyPin(enteredPin)) {
                            System.out.print("Enter Withdrawal Amount: ");
                            double withdrawalAmount = scanner.nextDouble();
                            scanner.nextLine(); // Consume the newline character
                            System.out.println("---------------");
                            withdrawAccount.withdraw(withdrawalAmount);
                            bank.saveAccounts(); // Save account balances after withdrawal
                            Bank.loadingScreen();
                        } else {
                            System.out.println("Invalid PIN.");
                            Bank.loadingScreen();
                        }
                    } else {
                        System.out.println("Account not found.");
                        Bank.loadingScreen();
                    }
                    break;
                case 4:
                    System.out.println("   _____ __              __             _    __            ____");
                    System.out.println("  / ___// /_  ____ _____/ /___ _      _| |  / /___ ___  __/ / /_");
                    System.out.println("  \\__ \\/ __ \\/ __ `/ __  / __ \\ | /| / / | / / __ `/ / / / / __/");
                    System.out.println(" ___/ / / / / /_/ / /_/ / /_/ / |/ |/ /| |/ / /_/ / /_/ / / /_");
                    System.out.println("/____/_/ /_/\\__,_/\\__,_/\\____/|__/|__/ |___/\\__,_/\\__,_/_/\\__/");
                    System.out.println("                                                By veilwr4ith             ");
                    System.out.println("\nAccount Details");
                    System.out.println("---------------");
                    System.out.print("Enter Account Number: ");
                    accountNumber = scanner.nextLine();
                    BankAccount accountDetailsAccount = bank.findAccount(accountNumber);
                    if (accountDetailsAccount != null) {
                        System.out.print("Enter PIN: ");
                        char[] enteredPin = System.console().readPassword(); // Read the PIN without echoing
                        System.out.println("---------------");
                        if (accountDetailsAccount.verifyPin(enteredPin)) {
                            accountDetailsAccount.displayAccountDetails();
                            Bank.loadingScreen();
                        } else {
                            System.out.println("Invalid PIN.");
                            Bank.loadingScreen();
                        }
                    } else {
                        System.out.println("Account not found.");
                        Bank.loadingScreen();
                    }
                    break;

                case 5:
                    System.out.println("   _____ __              __             _    __            ____");
                    System.out.println("  / ___// /_  ____ _____/ /___ _      _| |  / /___ ___  __/ / /_");
                    System.out.println("  \\__ \\/ __ \\/ __ `/ __  / __ \\ | /| / / | / / __ `/ / / / / __/");
                    System.out.println(" ___/ / / / / /_/ / /_/ / /_/ / |/ |/ /| |/ / /_/ / /_/ / / /_");
                    System.out.println("/____/_/ /_/\\__,_/\\__,_/\\____/|__/|__/ |___/\\__,_/\\__,_/_/\\__/");
                    System.out.println("                                                By veilwr4ith             ");
                    System.out.println("\nRequest Loan");
                    System.out.println("---------------");
                    System.out.print("Enter Account Number: ");
                    accountNumber = scanner.nextLine();
                    BankAccount loanAccount = bank.findAccount(accountNumber);
                    if (loanAccount != null) {
                        System.out.print("Enter PIN: ");
                        char[] enteredPin = System.console().readPassword(); // Read the PIN without echoing
                        if (loanAccount.verifyPin(enteredPin)) {
                            System.out.print("Enter Loan Amount: ");
                            double loanAmount = scanner.nextDouble();
                            scanner.nextLine(); // Consume the newline character
                            System.out.println("---------------");
                            bank.requestLoan(accountNumber, enteredPin, loanAmount);
                        } else {
                            System.out.println("Invalid PIN.");
                            Bank.loadingScreen();
                        }
                    } else {
                        System.out.println("Account not found.");
                        Bank.loadingScreen();
                    }
                    break;
                case 6:
                    System.out.println("   _____ __              __             _    __            ____");
                    System.out.println("  / ___// /_  ____ _____/ /___ _      _| |  / /___ ___  __/ / /_");
                    System.out.println("  \\__ \\/ __ \\/ __ `/ __  / __ \\ | /| / / | / / __ `/ / / / / __/");
                    System.out.println(" ___/ / / / / /_/ / /_/ / /_/ / |/ |/ /| |/ / /_/ / /_/ / / /_");
                    System.out.println("/____/_/ /_/\\__,_/\\__,_/\\____/|__/|__/ |___/\\__,_/\\__,_/_/\\__/");
                    System.out.println("                                                By veilwr4ith             ");
                    System.out.println("\nRepay Loan");
                    System.out.println("---------------");
                    System.out.print("Enter Account Number: ");
                    accountNumber = scanner.nextLine();
                    BankAccount repayLoanAccount = bank.findAccount(accountNumber);
                    if (repayLoanAccount != null) {
                        System.out.print("Enter PIN: ");
                        char[] enteredPin = System.console().readPassword(); // Read the PIN without echoing
                        if (repayLoanAccount.verifyPin(enteredPin)) {
                            System.out.print("Enter Repayment Amount: ");
                            double repaymentAmount = scanner.nextDouble();
                            scanner.nextLine(); // Consume the newline character
                            System.out.println("---------------");
                            repayLoanAccount.repayLoan(repaymentAmount);
                            bank.saveAccounts(); // Save account balances and loan details
                            Bank.loadingScreen();
                        } else {
                            System.out.println("Invalid PIN.");
                            Bank.loadingScreen();
                        }
                    } else {
                        System.out.println("Account not found.");
                        Bank.loadingScreen();
                    }
                    break;
                case 7:
                    System.out.println("   _____ __              __             _    __            ____");
                    System.out.println("  / ___// /_  ____ _____/ /___ _      _| |  / /___ ___  __/ / /_");
                    System.out.println("  \\__ \\/ __ \\/ __ `/ __  / __ \\ | /| / / | / / __ `/ / / / / __/");
                    System.out.println(" ___/ / / / / /_/ / /_/ / /_/ / |/ |/ /| |/ / /_/ / /_/ / / /_");
                    System.out.println("/____/_/ /_/\\__,_/\\__,_/\\____/|__/|__/ |___/\\__,_/\\__,_/_/\\__/");
                    System.out.println("                                                By veilwr4ith            ");
                    System.out.println("\nExiting the program. Thank you for using the Banking System!");
                    break;
                default:
                    System.out.println("Invalid choice. Please try again.");
                    Bank.loadingScreen();
            }
            if (choice != 7) {
                System.out.println();
                System.out.println("Press Enter to continue...");
                try {
                    System.in.read();
                    clearScreen();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }

        scanner.close();
    }

    public static void clearScreen() {
        try {
            if (System.getProperty("os.name").contains("Windows")) {
                new ProcessBuilder("cmd", "/c", "cls").inheritIO().start().waitFor();
            } else {
                System.out.print("\033[H\033[2J");
                System.out.flush();
            }
        } catch (IOException | InterruptedException e) {
            e.printStackTrace();
        }
    }
}